src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -96,98 +167,795 @@ static DamageTable D_80899510 = {
+        if (this->unk_2AE != 10) {
+            Math_StepToF(&this->unk_2C8, 0.0f, 0.05f);
+            if (this->unk_2AE == 0) {
+                this->unk_2CC = (this->unk_2C8 + 1.0f) * 1.05f;
+                this->unk_2CC = CLAMP_MAX(this->unk_2CC, 2.1f);
+            } else {
+                this->unk_2CC = (this->unk_2C8 + 1.0f) * 0.55f;
+                this->unk_2CC = CLAMP_MAX(this->unk_2CC, 1.1f);
+            }
+        } else if (!Math_StepToF(&this->unk_2D0, 1.1f, 0.0275f)) {
+            func_800B9010(thisx, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+        }
+    }
+}
+
+s32 EnPeehat_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                              Actor* thisx) {
+    EnPeehat* this = THIS;
+    s32 pad;
+
+    if (limbIndex == 4) {
+        rot->x = -this->unk_2B4;
+    } else if ((limbIndex == 3) ||
+               ((limbIndex == 23) && ((this->actionFunc == func_80898594) || (this->actionFunc == func_80897520)))) {
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        Gfx* gfx = POLY_OPA_DISP;
+
+        Matrix_StatePush();
+        Matrix_RotateStateAroundXAxis(this->unk_2BC * 0.115f);
+        Matrix_InsertYRotation_f(this->unk_2BC * 0.13f, MTXMODE_APPLY);
+        Matrix_InsertZRotation_f(this->unk_2BC * 0.1f, MTXMODE_APPLY);
+        Matrix_Scale(1.0f - this->unk_2C4, this->unk_2C4 + 1.0f, 1.0f - this->unk_2C4, MTXMODE_APPLY);
+        Matrix_InsertZRotation_f(-(this->unk_2BC * 0.1f), MTXMODE_APPLY);
+        Matrix_InsertYRotation_f(-(this->unk_2BC * 0.13f), MTXMODE_APPLY);
+        Matrix_RotateStateAroundXAxis(-(this->unk_2BC * 0.115f));
+
+        gSPMatrix(&gfx[0], Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(&gfx[1], *dList);
+
+        Matrix_StatePop();
+
+        POLY_OPA_DISP = &gfx[2];
+
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
+        return true;
+    }
+    return false;
+}
+
+void EnPeehat_PostLimbDraw(GlobalContext* globalCtx2, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f D_80899570[] = {
+        { 1300.0f, 1200.0f, 0.0f },
+        { 1300.0f, -1200.0f, 0.0f },
+        { 1300.0f, 0.0f, 1200.0f },
+        { 1300.0f, 0.0f, -1200.0f },
+    };
+    static s8 D_808995A0[] = {
+        -1, -1, -1, -1, -1, -1, -1, 0, -1, -1, 2, -1, -1, 4, -1, -1, 6, -1, -1, 8, -1, -1, 10, -1,
+    };
+    GlobalContext* globalCtx = globalCtx2;
+    EnPeehat* this = THIS;
+    s32 i;
+    s32 index = D_808995A0[limbIndex];
+    Gfx* gfx;
+
+    if (index != -1) {
+        Matrix_GetStateTranslationAndScaledX(2000.0f, &this->unk_2EC[index]);
+        Matrix_GetStateTranslationAndScaledX(4000.0f, &this->unk_2EC[index + 1]);
+    }
+
+    if (limbIndex == 4) {
+        Matrix_GetStateTranslationAndScaledZ(5500.0f, &this->unk_2D4[0]);
+        Matrix_GetStateTranslationAndScaledZ(-5500.0f, &this->unk_2D4[1]);
+        return;
+    }
+
+    if (limbIndex == 3) {
```

Comment:
```suggestion
    if (limbIndex == 4) {
        Matrix_GetStateTranslationAndScaledZ(5500.0f, &this->unk_2D4[0]);
        Matrix_GetStateTranslationAndScaledZ(-5500.0f, &this->unk_2D4[1]);
    } else if (limbIndex == 3) {
```
Since these if-statements are related, I think they should be combined into an if-else